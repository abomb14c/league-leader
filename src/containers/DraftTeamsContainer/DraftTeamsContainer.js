import React from 'react';
import { connect } from 'react-redux';
import DraftCard from '../../components/DraftCard/DraftCard';
import { Draggable } from 'react-drag-and-drop';
import PropTypes from 'prop-types';

export const DraftTeamsContainer = (props) => {
  const displayDraftTeams = props.draftTeams.map((team, index) => {
    return (
      <Draggable type="team" data={team.name} key={index} >
        <DraftCard name={team.name}  />
      </Draggable>
    );
  });

  return (
    <div>
      {displayDraftTeams}
    </div>
  );
};


export const mapStateToProps = state => ({
  draftTeams: state.draftTeams
});

export default connect(mapStateToProps, null)(DraftTeamsContainer);

DraftTeamsContainer.propTypes = {
  draftTeams: PropTypes.arrayOf(PropTypes.object)
};