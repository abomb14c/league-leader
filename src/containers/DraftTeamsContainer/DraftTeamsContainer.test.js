import React from 'react';
import { shallow } from 'enzyme';
import {DraftTeamsContainer, mapStateToProps} from './DraftTeamsContainer';

describe('DraftTeamsContainer', () => {
  let wrapper;
  let mockProps; 

  beforeEach(() => {
    mockProps = {
      draftTeams: [{}, {}, {}]
    };

    wrapper = shallow(<DraftTeamsContainer
      {...mockProps}
    />);
  });
  
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});


describe('mapStateToProps', () => {
  it('should map draftTeams to props', () => {
    const mockState = {
      EPL: [{}, {}, {}],
      NBA: {'EASTERN CONFERENCE':[{}, {}, {}],
        'WESTERN CONFERENCE': [{}, {}, {}]
      },
      draftTeams: [{}, {}, {}]
    };

    const expected = {
      draftTeams: [{}, {}, {}]

    };
    
    const mappedProps = mapStateToProps(mockState);
    expect(mappedProps).toEqual(expected);
  });
});