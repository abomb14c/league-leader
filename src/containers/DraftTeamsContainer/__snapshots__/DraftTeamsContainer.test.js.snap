// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DraftTeamsContainer should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    draftTeams={
        Array [
            Object {},
            Object {},
            Object {},
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Draggable
          data={undefined}
          enabled={true}
          type="team"
>
          <Unknown
                    name={undefined}
          />
</Draggable>,
        <Draggable
          data={undefined}
          enabled={true}
          type="team"
>
          <Unknown
                    name={undefined}
          />
</Draggable>,
        <Draggable
          data={undefined}
          enabled={true}
          type="team"
>
          <Unknown
                    name={undefined}
          />
</Draggable>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "class",
        "props": Object {
          "children": <Unknown
            name={undefined}
/>,
          "data": undefined,
          "enabled": true,
          "type": "team",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "name": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "class",
        "props": Object {
          "children": <Unknown
            name={undefined}
/>,
          "data": undefined,
          "enabled": true,
          "type": "team",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "name": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "class",
        "props": Object {
          "children": <Unknown
            name={undefined}
/>,
          "data": undefined,
          "enabled": true,
          "type": "team",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "name": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Draggable
            data={undefined}
            enabled={true}
            type="team"
>
            <Unknown
                        name={undefined}
            />
</Draggable>,
          <Draggable
            data={undefined}
            enabled={true}
            type="team"
>
            <Unknown
                        name={undefined}
            />
</Draggable>,
          <Draggable
            data={undefined}
            enabled={true}
            type="team"
>
            <Unknown
                        name={undefined}
            />
</Draggable>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "class",
          "props": Object {
            "children": <Unknown
              name={undefined}
/>,
            "data": undefined,
            "enabled": true,
            "type": "team",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "name": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "children": <Unknown
              name={undefined}
/>,
            "data": undefined,
            "enabled": true,
            "type": "team",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "name": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "class",
          "props": Object {
            "children": <Unknown
              name={undefined}
/>,
            "data": undefined,
            "enabled": true,
            "type": "team",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "name": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
